### DISCUSSION TITLE
How can I integrate tw-elements?

### DISCUSSION THREAD

**Discussion by bangrobe** [gregsullivan/_tw#69] [General]
tw-elements.com just released their stable version. How can I integrate it into the theme? I think there are a way to do with esbuild, but i'm not a pro.
Thank you.

**Comment by gregsullivan** [comment]
Thanks for posting this—I'm going to convert this to a discussion, and we can continue talking about it there!

After a quick look at TW Elements, it seems like it's a component library with a JavaScript requirement for some components.

My take here is that you'll generally copy the HTML for the components you want to use directly into WordPress templates, but you're not sure how to handle the JavaScript requirement.

If that's the case, you should just need to run this:

`npm install tw-elements`

Then add their plugin to your `tailwind.config.js` file:

`require("tw-elements/dist/plugin.cjs")`

Then add these lines to your `javascript/script.js` file:

```javascript
import { Datepicker, Input, initTE } from "tw-elements";
initTE({ Datepicker, Input }, { allowReinits: true });
```

I'm not sure whether this varies depending on the components you use; I'm taking this from their documentation here:

https://tw-elements.com/docs/standard/getting-started/quick-start/

Hope that helps!
